### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by Hyperkid123**
I've documented a couple of components in order to debug the development process

### Fixes
- escape unsafe characters only for stings (JSDOC crashed on default values which were numbers)
- add examples folder to `generate:watch`

### Docs
- added docs for DateFormat component
- added docs for CheckboxFilter component
- fix optional arguments flag and default values for debounce function

screenshot-localhost_3000-2021 03 04-16_06_46

### CODE REVIEW

File: packages/components/src/ConditionalFilter/CheckboxFilter.js [hunk 1]
```diff
@@ -4,7 +4,11 @@ import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
 import Text from './TextFilter';
 import isEqual from 'lodash/isEqual';
 
-class Checkbox extends Component {
+/**
+ * CheckboxFilter is a part of the <a href="/components/ConditionalFilter">Conditional filter composite component</a>.
+ * It was not designed to be used as a standlone component.
```
Comment (rvsia) [inline packages/components/src/ConditionalFilter/CheckboxFilter.js:9]: typo: standalone

File: packages/docs/examples/DateFormat/data-types.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+
+const DifferentDateDataTypes = () => {
+    return (
```
Comment (rvsia) [inline packages/docs/examples/DateFormat/data-types.js:5]: unnecessary block statement

File: packages/docs/examples/DateFormat/data-types.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        <div>
+            <span>Using Date object: &nbsp;</span><DateFormat date={new Date()} />< br />
+            <span>Using number: &nbsp;</span><DateFormat date={Date.now()} />< br />
+            <span>Using date string: &nbsp;</span><DateFormat date={new Date().toString()} />< br />
+        </div>
+    );
+};
+
+export  default DifferentDateDataTypes;
```
Comment (rvsia) [inline packages/docs/examples/DateFormat/data-types.js:14]: two spaced

File: packages/docs/examples/DateFormat/extra-title.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+
+const date = new Date();
+const extraStringTitle = 'Extra string title ';
+const extraNodeTitle = <h2>Extra node title</h2>;
+
+const DateFormats = () => {
+    return (
```
Comment (rvsia) [inline packages/docs/examples/DateFormat/extra-title.js:9]: unnecessary block statement

File: packages/docs/examples/DateFormat/extra-title.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        <div>
+            <span>Extra tooltip title as string: &nbsp;</span><DateFormat extraTitle={extraStringTitle} date={date} />< br />
+            <span>Extra tooltip title as react node: &nbsp;</span><DateFormat extraTitle={extraNodeTitle} date={date} />< br />
+        </div>
+    );
+};
+
+export  default DateFormats;
```
Comment (rvsia) [inline packages/docs/examples/DateFormat/extra-title.js:17]: Oh, I think you copied the two spaces in multiple files. 😄

File: packages/docs/examples/DateFormat/tooltip-props.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const DifferentDateDataTypes = () => {
+    const date = new Date();
+    const tooltipProps = {
+        position: 'left',
+        distance: 80
+    };
+    return (
+        <div>
+            <Title size="lg" headingLevel="h3">
+                Tooltip shows on the left size and is further from the root element.
```
Comment (rvsia) [inline packages/docs/examples/DateFormat/tooltip-props.js:14]: is shown on the left side

**Comment by Hyperkid123** [comment]
Oh, the typos. I think its time for some grammar check vs code extension :D

File: packages/docs/examples/DateFormat/data-types.js (hunk 3 above)
Comment (Hyperkid123) [reply packages/docs/examples/DateFormat/data-types.js:14]: Wonder why eslint is not catching this

**Review by rvsia — APPROVED** [review]

**Pull request merged**